### PR TITLE
MULE-19361: MULE-19361: add ParameterGroupModel as key for Parameters map

### PR DESCRIPTION
**Pull request by jcgrethe**

### CODE REVIEW

**Comment by jcgrethe** [comment]
--validate

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/config/MultipleParametersGroupTestCase.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    return new String[] {"heisenberg-multiple-parameters-group-config.xml"};
+  }
+
+  @Override
+  protected boolean isDisposeContextPerClass() {
+    return true;
+  }
+
+  @Test
+  public void getInlineGroupDefinition() throws Exception {
```
Comment (elrodro83) [inline modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/config/MultipleParametersGroupTestCase.java:39]: allure issue annotation

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/config/MultipleParametersGroupTestCase.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+public class MultipleParametersGroupTestCase extends AbstractExtensionFunctionalTestCase {
+
+  public static final String HEISENBERG = "heisenberg";
+  private static final String EMPTY_STRING = "";
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Override
+  protected String[] getConfigFiles() {
```
Comment (elrodro83) [inline modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/config/MultipleParametersGroupTestCase.java:29]: why not use `getConfigFile`?

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/config/MultipleParametersGroupTestCase.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    return new String[] {"heisenberg-multiple-parameters-group-config.xml"};
+  }
+
+  @Override
+  protected boolean isDisposeContextPerClass() {
+    return true;
+  }
```
Comment (elrodro83) [inline modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/config/MultipleParametersGroupTestCase.java:36]: remove this, doesn't seem to be needed

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/config/MultipleParametersGroupTestCase.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+  @Test
+  public void getInlineGroupDefinition() throws Exception {
+    Message message = flowRunner("getBarberPreferences").withPayload(EMPTY_STRING).run().getMessage();
```
Comment (elrodro83) [inline modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/config/MultipleParametersGroupTestCase.java:40]: do you actually need to set the empty payload?

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by elrodro83 — APPROVED** [review]

**Comment by jcgrethe** [comment]
--validate

**Pull request merged**